hnn_core/network.py

```diff
@@ -220,11 +227,26 @@ def copy(self):
             A copy of the instance with previous simulation results and
             ``events`` of external drives removed.
         """
+        # reset cells to avoid pickling error
+        self.update_cells()
         net_copy = deepcopy(self)
         net_copy.cell_response = CellResponse(times=self.cell_response._times)
         net_copy._reset_drives()
         return net_copy
 
+    def update_cells(self):
+        '''Populate the network with cell objects'''
+
+        for cell_type in self.pos_dict.keys():
+            if cell_type in self.cell_types:
+                cells = list()
+                for cell_idx, pos in enumerate(self.pos_dict[cell_type]):
+                    cell = deepcopy(self.cell_types[cell_type])
+                    cell.gid = self.gid_ranges[cell_type][cell_idx]
+                    cell.pos = pos
+                    cells.append(cell)
+                    self.cells[cell_type] = cells
+
```

Comment:
This is where `Cell` instances are replicated to produce the cell population stored in `Network.cells`. We can leave this here for now or I can move it over to `NetworkBuilder`. Any preference @jasmainak @ntolley?